This branch is to add the Feat: Create book and Remove book using redux

steveWDamesJr opened the pull request:
## Add Functionality - Create book and Remove Book

-Configure Redux Store
-Write actions and Reducers for books
-Set Data inputs in the Local React state
-Dispatch corresponding actions in deleteBook
-Implement AddBook Logic
-Implement Remove Book Logic
-Check application with Redux-Logger

-Debug Linters

steveWDamesJr committed:
Initialize AddBook component

steveWDamesJr committed:
Re-initialize AddBook component

-Configure Redux Store
-Write actions and Reducers for books
-Set Data inputs in the Local React state
-Dispatch corresponding actions in deleteBook
-Implement AddBook Logic
-Implement Remove Book Logic
-Check application with Redux-Logger

AnselemOdims reviewed (changes requested):
### Hi @steveWDamesJr, 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [x] Please include a redux configuration(action type, action creators, and reducer) for the `categories.js` as per the requirements. At the moment, the file is empty.

## Optional suggestions
- [x] Please always ensure that a milestone is only merged when approved. This will enable the reviewer to see all current changes related to that milestone. In this case, I had to check the default branch to see the changes made.

_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

steveWDamesJr committed:
Add categories actions and Reducers

-Define an action type for checking the status.
-Set the initial state to be an empty array of categories.

-Export Action Creators for your actions.
-Write your reducer and export it as default.
-For the check status action return a string Under construction.
-In case of unknown action - return the current state.

steveWDamesJr committed:
Merge branch 'dev' into create-remove-books

steveWDamesJr committed:
Add categories actions and Reducers

-Define an action type for checking the status.
    -Set the initial state to be an empty array of categories.
    -Export Action Creators for your actions.
    -Write your reducer and export it as default.
    -For the check status action return a string Under construction.
    -In case of unknown action - return the current state.2

elyor-doniyorov reviewed (approved):
Hi @steveWDamesJr, 

In my opinion, your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
